src/libraries/System.Text.Json/tests/Common/ConstructorTests/ConstructorTests.ParameterMatching.cs

```diff
@@ -1636,5 +1638,33 @@ public class CustomCtorParameterConverter : JsonConverter<string>
             public override void Write(Utf8JsonWriter writer, string value, JsonSerializerOptions options)
                 => writer.WriteStringValue(value);
         }
+
+        [Fact]
+        public async Task RespectRequiredConstructorParameters_RequiredParameterMissing_ThrowsJsonException()
+        {
+            string json = """{"X":1,"Z":3}""";
+            JsonException ex = await Assert.ThrowsAsync<JsonException>(() => Serializer.DeserializeWrapper<Point_3D>(json, s_respectRequiredParamsOptions));
+            Assert.Contains("was missing required properties including: 'Y'.", ex.Message);
```

Comment:
I vaguely remember @danmoseley opposed to test for parts or whole error messages, except for the bits that were immune to localization e.g. the single quoted name of properties in this case.

Reply:
```suggestion
            Assert.Contains("'Y'", ex.Message);
```